### ISSUE TITLE
[BUG][Fuzzer][wrong-code] Vectorized broadcast-store of a negative `int32` across a 256-bit vector silently writes the sign-extension (not the value) to half the lanes on Blackwell (sm_100)

### ISSUE THREAD

**Issue by Hughshine** [tile-ai/tilelang#2489]
### Required prerequisites

- [x] I have read the documentation <https://tilelang.com>.
- [x] I have searched the [Issue Tracker](https://github.com/tile-ai/tilelang/issues) that this hasn't already been reported. (comment there if it has.)

### What version of TileLang are you using?

0.1.11

### System information

- Installed from the `0.1.11` wheel; source pinned to tag `v0.1.11` (commit `cd37ed5fc35ae7a60a1277c8eb49028174ac51e6`). `apache-tvm-ffi==0.1.11`.
- Python 3.13, linux; CUDA 13.0, driver 580.x.
- GPU: **NVIDIA B200 (sm_100, Blackwell)**. The defect is architecture-gated to sm_100 (the 256-bit vectorized store is enabled only for `arch >= 100`); it does **not** occur on sm_90 / Hopper and earlier, where the path is unreachable. **Reproduced on B200 on 2026-06-27; not re-run this session (no sm_100 hardware currently available) — see Verification in Additional context.**

### Problem description

A `T.Parallel` vectorized broadcast-store of a **negative signed `int32`** across a vector wide enough to trigger 256-bit vectorization (8 `int32` lanes) **silently writes the sign-extension (e.g. `-1`) to the high lane of every packed pair instead of the value**. Exactly half the elements come out wrong, with no compile error, no runtime error, and no out-of-bounds. The codegen packs two `int32` lanes per 64-bit slot as `((long long)(int)v) | ((long long)(int)v << 32)`; for a negative `v` the `(int)v` widened to `long long` sign-extends, so the high 32 bits become the sign (all-ones) rather than a second copy of `v`. The `is_uint()` sibling widens through `unsigned` (zero-extend) and is correct — the defect is signed-only.

<details>
<summary>Root cause — signed widening sign-extends into the high lane (uint sibling zero-extends)</summary>

The 256-bit `int32` pack appears in two handlers in `src/cuda/codegen/codegen_cuda.cc`, both with the identical formula. BroadcastNode (the `bits()==32 && lanes>4` branch, guard at [`#L5057`](https://github.com/tile-ai/tilelang/blob/cd37ed5fc35ae7a60a1277c8eb49028174ac51e6/src/cuda/codegen/codegen_cuda.cc#L5057), pack at [`#L5069`](https://github.com/tile-ai/tilelang/blob/cd37ed5fc35ae7a60a1277c8eb49028174ac51e6/src/cuda/codegen/codegen_cuda.cc#L5069)) and ShuffleNode ([`#L4943`](https://github.com/tile-ai/tilelang/blob/cd37ed5fc35ae7a60a1277c8eb49028174ac51e6/src/cuda/codegen/codegen_cuda.cc#L4943)):

```cpp
const char *u64 = op->dtype.is_uint() ? "unsigned long long" : "long long";
const char *u32 = op->dtype.is_uint() ? "unsigned int" : "int";
...
os << "((" << u64 << ")(" << u32 << ")(" << v << ")) | "
   << "((" << u64 << ")(" << u32 << ")(" << v << ") << 32)";
```

For signed `int32`, `(int)v` widened to `long long` **sign-extends**: a negative `v` makes the high 32 bits all-ones, corrupting the upper lane of each pair. The emitted code for the repro is:

```cpp
tl::store_global_256(&(*(longlong4*)(A + ...)),
    make_longlong4(((long long)(int)(broadcast_var)) | ((long long)(int)(broadcast_var) << 32), ...));
```

The fix mirrors the already-correct `unsigned` branch: mask (`& 0xFFFFFFFF`) or widen through `unsigned` before the 64-bit shift.
</details>

<details>
<summary>Why it is architecture-gated to Blackwell (sm_100)</summary>

This pack is only emitted when the store is 256-bit-vectorized (`lanes > 4`), and 256-bit vectorization is enabled only for `arch >= 100`: [`TargetSupportVectorize256`](https://github.com/tile-ai/tilelang/blob/cd37ed5fc35ae7a60a1277c8eb49028174ac51e6/src/backend/common/target_utils.cc#L168) returns `arch >= 100`. On sm_90 (H100) and earlier the path is unreachable — every kernel vectorizes to ≤128-bit / x4 and never emits the 64-bit pack (verified by exhausting the attempts on H100). So this is a Blackwell-only silent miscompile.
</details>

### Reproducible example code

Use a real `.py` file. Run on a B200 / sm_100.

```python
import tilelang, tilelang.language as T, torch
M, npt = 2048, 128
@T.prim_func
def main(A: T.Tensor((M,), "int32")):
    with T.Kernel(T.ceildiv(M, npt), threads=1) as bx:
        for j in T.Parallel(npt):
            idx = bx * npt + j
            if idx < M:
                A[idx] = T.int32(-123456789)        # broadcast a NEGATIVE int32
mod = tilelang.compile(main, target="cuda")          # on a B200 / sm_100
A = torch.zeros(M, dtype=torch.int32, device="cuda"); mod(A)
print((A != -123456789).sum().item(), "/", M, "wrong; sample:", A[A != -123456789][:4].tolist())
# -> 1024 / 2048 wrong; sample: [-1, -1, -1, -1]   (the high lane of each packed pair)
```

The control isolates the cause to the signed cast: the identical kernel with `uint32` / `T.uint32(0xF8A432EB)` gives **0 / 2048 wrong** — only the signed path miscompiles.

### Traceback

```pytb
No traceback — the kernel compiles and runs to completion; exactly half the elements (the high lane of each packed pair) are silently -1 instead of the value, deterministically.
```

### Expected behavior

The broadcast value should be stored unchanged in every lane. The signed 256-bit pack must zero-extend the low 32 bits (mask `& 0xFFFFFFFF`, or widen through `unsigned`) exactly as the `is_uint()` branch already does — a one-line change mirroring the existing correct sibling.

### Additional context

<details>
<summary>It is sign-extension, NOT overflow (B200 diagnostic, 2026-06-27)</summary>

| stored value | #wrong | low lanes wrong | high lanes wrong | wrong value |
|---|---|---|---|---|
| `-123456789` | 1024 | 0 | 1024 | `-1` |
| `-2` | 1024 | 0 | 1024 | `-1` |
| `-2147483648` (INT_MIN) | 1024 | 0 | 1024 | `-1` |
| `-1` | **0** | 0 | 0 | — |
| `2147483647` (INT_MAX) | **0** | 0 | 0 | — |
| `123456789` | 0 | 0 | 0 | — |

Only the high lane of each pair is wrong; the wrong value is always `-1` regardless of the magnitude of `v`; every non-negative value (including INT_MAX) is correct; and `v = -1` is silently *correct* because its sign-extension `0xFFFFFFFF` coincidentally equals the value it should store. These are the fingerprints of sign-extension, not arithmetic overflow.
</details>

**Verification status (read this before filing).** Reproduced on **NVIDIA B200 (sm_100), 2026-06-27** — signed → 1024/2048 silently `-1`, uint control → 0/2048 — with the emitted source confirmed to contain the `((long long)(int)…)|(…<<32)` pack (`validator/tilelang/bytelevel/evidence/sm100_int32x8_repro.py` + uint control). The bit-formula was independently proven wrong by a standalone host unit test (`evidence/int32x2_pack64_signed_unit.cpp`, exit=1: `v=-123456789 → lane1=-1`). The source mechanism and the `arch >= 100` reachability gate are **re-verified against live `main` this session** (the pack at codegen_cuda.cc:4943/5069 and the gate at target_utils.cc:168 are unchanged). **NOT re-run on silicon this session — no sm_100 hardware currently available; the silicon repro above is from 2026-06-27.** Fix not built.

**Provenance.** The signed `((long long)(int)v)|(…<<32)` pack was introduced by [#2093](https://github.com/tile-ai/tilelang/pull/2093) ("Add full Windows support and fix related cross-platform issues", 2026-05-06) — the same PR is the introducing commit for both the BroadcastNode and ShuffleNode forms (`git log -S` on the pack expression returns only that commit). The 256-bit vectorization gate `TargetSupportVectorize256` (`arch >= 100`) predates it (introduced by [#1644](https://github.com/tile-ai/tilelang/pull/1644), 2026-01-18). So the defect dates to #2093's pack codegen, gated to sm_100 by the earlier vectorization arch check.
